RNWCPP/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +30,283 @@ enum class ScrollViewCommands
+    {
+      double x = commandArgs[0].asDouble();
+      double y = commandArgs[1].asDouble();
+      bool animated = commandArgs[2].asBool();
+      bool horiz = scrollViewer.HorizontalScrollMode() == winrt::ScrollMode::Auto;
+      if (horiz)
+      {
+        // snap the x coordinate
+        if (m_isSnapToAlignmentSet && m_snapToInterval > 0.0)
+        {
+          x = snap(x, m_snapToAlignment, m_snapToInterval, scrollViewer.ViewportWidth());
+        }
+        // ChangeView parameters: x, y, zoom, animated
+        scrollViewer.ChangeView(x, nullptr, nullptr, !animated/*disableAnimation*/);
+      }
+      else
+      {
+        // snap the y coordinate
+        if (m_isSnapToAlignmentSet && m_snapToInterval > 0.0)
+        {
+          y = snap(y, m_snapToAlignment, m_snapToInterval, scrollViewer.ViewportHeight());
+        }
+        // ChangeView parameters: x, y, zoom, animated
+        scrollViewer.ChangeView(nullptr, y, nullptr, !animated/*disableAnimation*/);
+      }
+      break;
+    }
+
+    case static_cast<int64_t>(ScrollViewCommands::ScrollToEnd) :
+    {
+      bool animated = commandArgs[0].asBool();
+      bool horiz = scrollViewer.HorizontalScrollMode() == winrt::ScrollMode::Auto;
+      if (horiz)
+        scrollViewer.ChangeView(scrollViewer.ScrollableWidth(), nullptr, nullptr, !animated/*disableAnimation*/);
+      else
+        scrollViewer.ChangeView(nullptr, scrollViewer.ScrollableHeight(), nullptr, !animated/*disableAnimation*/);
+      break;
+    }
+  }
+}
+
+void ScrollViewShadowNode::createView()
+{
+  Super::createView();
+
+  auto scrollViewer = GetView().as<winrt::ScrollViewer>();
+  ScrollViewManager* viewManager = static_cast<ScrollViewManager*>(GetViewManager());
+  int64_t tag = m_tag;
+
+  scrollViewer.ViewChanging([this, tag](const winrt::IInspectable& sender, const winrt::ScrollViewerViewChangingEventArgs& args)
+  {
+    winrt::ScrollViewer scrollViewer = sender.as<winrt::ScrollViewer>();
+    topScroll(scrollViewer, tag,
+      "topScroll",
+      args.NextView().HorizontalOffset(),
+      args.NextView().VerticalOffset(),
+      args.NextView().ZoomFactor());
+
+  });
+}
+ 
+
+void ScrollViewShadowNode::updateProperties(const folly::dynamic&& props)
+{
+  m_updating = true;
+
+  auto scrollViewer = GetView().as<winrt::ScrollViewer>();
+  if (scrollViewer == nullptr)
+    return;
+
+  for (auto& pair : props.items())
+  {
+    const folly::dynamic& propertyName = pair.first;
+    const folly::dynamic& propertyValue = pair.second;
+
+    if (propertyName.asString() == "horizontal")
+    {
+      if (propertyValue.isBool())
+      {
+        bool horiz = propertyValue.getBool();
+
+        scrollViewer.HorizontalScrollMode(horiz ? winrt::ScrollMode::Auto : winrt::ScrollMode::Disabled);
+        scrollViewer.VerticalScrollMode(horiz ? winrt::ScrollMode::Disabled : winrt::ScrollMode::Auto);
+      }
+    }
+    else if (propertyName.asString() == "showsHorizontalScrollIndicator")
```

Comment:
>showsHorizontalScrollIndicator [](start = 41, length = 30)

How are you testing this?  I don't see anything in ScrollViewSimpleExample.js that tests this particular property.  Same question applies to the other properties.  The ask is to make sure something in RNTester exercises all the properties you're adding.

Reply:
note most of the changes in this file are moving a function from ScrollViewManager::UpdateProperties to ScrollViewShadowNode::updateProperties.  Chris didn't implement most of that its just what we had before which was some initial guessing at props

Reply:
Fair enough...  Chris, feel free to resolve anything pre-existing as "By Design", but please file an issue in GitHub to track (one issue with a copy/paste of my open questions would be fine).

---
In reply to: [273329031](https://github.com/Microsoft/react-native-windows/pull/2263#discussion_r273329031) [](ancestors = 273329031)